src/mat/impls/inv/matinv.c

```diff
@@ -79,10 +94,11 @@ static PetscErrorCode MatInvComputeNullSpace_Inv(Mat imat)
     TRY( PCFactorGetMatSolverType(pc,&type) );
     TRY( PetscStrcmp(type,MATSOLVERMUMPS,&flg) ); 
     if (flg) TRY( PetscObjectTypeCompare((PetscObject)pc,PCCHOLESKY,&flg) );
+    //TODO what prevents us to use MatMumpsSet* API in either case
```

Comment:
You need to call PCSetUP() (which does the factorization) before being able to call PCFactorGetMatrix().

Reply:
OK, I changed the comment.